### DISCUSSION TITLE
Add page label support to PdfWriter

### DISCUSSION THREAD

**Discussion by lorenzomanini** [py-pdf/pypdf#1554] [Ideas]
Hello,
I find this project very useful since I like to add features to PDFs on which I study: thank you for keeping it alive.
I noticed in the latest version reading page labels was added(#1519), so I wanted to encourage further development, including adding an add_page_labels method to PdfWriter.
What do you think?

**Comment by MartinThoma** [comment]
I'm open to PRs adding page label support to `PdfWriter` :-)

The first question would be about the interface. Before we decide on a public interface, we could add a method to PdfWriter:

```python
def _set_page_label(
    self,
    page_index_from: int,
    page_index_to: int,
    label: Literal["/D", "/R", "/r", "/A", "/a"],
):
    ...
```

Flee free to add a PR :-)

**Reply by lorenzomanini** [reply]
Good. I have never contributed to an open-source project, and my experience with python is limited, but I'd like to give it a shot.
Thank you for the response and the hint: I'll see what I can do!
